New effect base

greatguys1 opened the pull request.

greatguys1 committed:
Rewrote base_effect and created base_dot

- Effect base handles registering simple effects and player actions (like sitting). Much smaller and easier to work with.

- Dot Base handles damage over time effects, resistance checks, and has additional functions to help set up visual fx.
-- DoTs now support stacking
--- Functional, but still needs some work

- Adjusted all fire effects to work with new system

greatguys1 committed:
More effect script stuff

- tbh i kinda forgor what I did :skull:
- Web effect near complete, now todo cocoon and vfx

greatguys1 committed:
Fix bug where elite bandit could end being frozen early

greatguys1 committed:
Remove soon-to-be defunct scripts

greatguys1 committed:
Finish web effect

- Most monsters can be webbed
- Amt of web is dependent on size of monster

greatguys1 committed:
Balance webs + web fixes

greatguys1 committed:
Adjust bleed dot to use new base dot

greatguys1 committed:
Fix client script improperly being removed

greatguys1 committed:
Acid dot converted to use new base dot

greatguys1 committed:
Defile dot converted to use new dot base

greatguys1 committed:
Convert disease for new base dot + fixes

- Fix some fx not clearing when a dot expires early
- Fix base dot not showing afflicted text

greatguys1 committed:
Change drunk effect to be entirely client side

greatguys1 committed:
Remove unnecessary script

greatguys1 committed:
Converted cold dot to new base dot

- Removed many redundant scripts relating to other cold dots

greatguys1 committed:
Convert ice shield & holy dot to new effect bases

greatguys1 committed:
Remove unused script

greatguys1 committed:
Remove more unused scripts

greatguys1 committed:
Convert lightning dot for new base dot

greatguys1 committed:
Convert "nojump" effect to new base effect

greatguys1 committed:
Convert poison effect to new base dot

greatguys1 committed:
Convert effect_push + effect_quake 4 new base effect

greatguys1 committed:
Convert rejuv effect for new base effect

greatguys1 committed:
Convert shadowflames to dot_dark w new base

greatguys1 committed:
Convert effect_rejuv2 to new base

greatguys1 committed:
Convert shardshield to new base

greatguys1 committed:
Shadowflame name changes

greatguys1 committed:
Adjust dots to be less dependent on client side scripts

greatguys1 committed:
Convert slow, spiderlatch, stam regen, & stop effects to new base

greatguys1 committed:
Convert stun to new base

- No more heavy stun variant, all stuns prevent players from attacking.

greatguys1 committed:
convert templock for new effect base

greatguys1 committed:
convert tempnomove for new base

greatguys1 committed:
convert force cage to new base

greatguys1 committed:
proper glow shell around poison spell proj

greatguys1 committed:
Remove freeze_ray and freeze_slow

- Unused / bugged

greatguys1 committed:
Remove generic_damage

- Unnecessary script, only 1 use case which was easily replaced.

greatguys1 committed:
Convert gauntlet_invalid for new base

greatguys1 committed:
Update freeze_solid for new base

- Adds a function to base_dot, dot_check_canapply, which will also call that function in higher scripts, in case they need additional functions to prevent application

greatguys1 committed:
Convert goblin_latch to new base

greatguys1 committed:
Convert greater poison for new dot base

- Players can now be afflicted with the damage reduction

greatguys1 committed:
Convert hold_person to new base dot

greatguys1 committed:
Made scarabs evil again

greatguys1 committed:
convert 2h axe red charge to new base dot

greatguys1 committed:
Convert haste for new base effect

greatguys1 committed:
make effects/speed obsolete and removed it

greatguys1 committed:
convert torch_flame to new effect base

greatguys1 committed:
remove effects/speed from precache

greatguys1 committed:
Remove unused script

greatguys1 committed:
Pt1 using scriptflags to track dots

greatguys1 committed:
Merge branch 'new-effect-base' of https://github.com/MSRevive/MSCScripts into new-effect-base

greatguys1 committed:
Finish using scriptflags to track dots

greatguys1 committed:
New event for scriptflag update to help higher scripts

greatguys1 committed:
oops

greatguys1 committed:
Pt1 New base debuff + add function for time left on base effect

greatguys1 committed:
Start moving debuffs to utilize base_debuff script

greatguys1 committed:
Create debuff_lightning, remove greater poison, start work on debuff_freeze

greatguys1 committed:
Finish freeze for base debuff

greatguys1 committed:
Move stun to base_debuff

greatguys1 committed:
add base_debuff_diminishing + wip on converting some effects over

greatguys1 committed:
WIP more hold debuff stuff

greatguys1 committed:
Finish converting effects to new bases

greatguys1 committed:
Add status icons to effects

greatguys1 committed:
Remove applyeffectstack

greatguys1 committed:
Fix maldora minions dot refreshing its own dot

greatguys1 committed:
Fix many monsters applying endless DoTs

greatguys1 committed:
Fix fire fx going nuts when reapplied + add dev npc

- dev npc is just a wall with tons of hp

greatguys1 committed:
Fix unclosed brace

greatguys1 committed:
Remove many unused vars

greatguys1 committed:
Add dev staff

greatguys1 committed:
Fix holy dot dotting twice

greatguys1 committed:
Fix dots effecting certain npcs only once

greatguys1 committed:
Fix more enemies doing infinite dots

greatguys1 committed:
Reduce max stun/frozen time to 10 seconds

greatguys1 committed:
More dev staff features

greatguys1 committed:
More effects system stuff

- Remove usage of IMMUNE_STUN, IMMUNE_PUSH, and MSC_PUSH_RESIST and replaced it with "takedmg stun x".
- Removed improper usage of takedmg ice
- Properly reduce debuff max time

Jaggernaut555 commented on scripts/effects/dot_acid.script:
```diff
@@ -0,0 +1,61 @@
+
+{ game_damaged
+
+	//take 150% damage while under this effect
```
This applies the acid x1.5 effect and as well as the `debuff_acid` effect that does another x1.5 multiplier.

I was messing around with some acid stuff in this branch by total chance and noticed that it wasn't doing the expected damage lol

greatguys1 replied:
Oh, thats not intended. Thank you for letting me know

greatguys1 committed:
Fix acid dot from increasing damage more than usual

greatguys1 committed:
Fix a few bugs with effects

- Can no longer apply dots on skeletons that are playing dead
- Fix skeletons sometimes being unaffected by a dot that killed them, after reviving
- NPCs can no longer afflict players multiple times with the same effect

greatguys1 committed:
Fix stun resist application and display

greatguys1 committed:
More fixes

- Fix no stun immunity being full stun immunity.
- Fix bluntarms not being able to stun

greatguys1 committed:
Merge branch 'MSR-Alpha' into new-effect-base

greatguys1 committed:
Update info on ms.stx, slightly buff blind in poison

greatguys1 committed:
Remove IS_LURE var
